kotest-assertions/kotest-assertions-json/src/commonMain/kotlin/io/kotest/assertions/json/wrappers.kt

```diff
@@ -0,0 +1,28 @@
+package io.kotest.assertions.json
+
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.booleanOrNull
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.doubleOrNull
+import kotlinx.serialization.json.floatOrNull
+import kotlinx.serialization.json.intOrNull
+import kotlinx.serialization.json.longOrNull
+
+fun JsonElement.toJsonNode(): JsonNode = when (this) {
+   JsonNull -> JsonNode.NullNode
+   is JsonObject -> JsonNode.ObjectNode(entries.map { it.key to it.value.toJsonNode() }.toMap())
+   is JsonArray -> JsonNode.ArrayNode(map { it.toJsonNode() })
+   is JsonPrimitive -> when {
+      intOrNull != null -> JsonNode.IntNode(intOrNull!!)
```

Comment:
What is intOrNull here? Looks like an imported function ?

Reply:
Yeah these are extension methods on the `JsonPrimitive` type that unwrap the value for you

Reply:
Oh wow, so you're in a receiver and using extensions. That's cool.